chain/store/snapshot.go

```diff
@@ -113,13 +130,17 @@ func (cs *ChainStore) Import(ctx context.Context, r io.Reader) (*types.TipSet, e
 	// check errors
 	for i := 0; i < parallelPuts; i++ {
 		if lastErr := <-putThrottle; lastErr != nil {
-			return nil, lastErr
+			return nil, nil, lastErr
 		}
 	}
 
+	if tailBlock.Height != 0 {
```

Comment:
What about the json genesis thing? (I assume you've tested this, I just wanted to make sure we're not missing something)

Reply:
The first block with the BlockHeader type is at height 0, so here we just check for that - https://filfox.info/en/block/bafy2bzacecnamqgqmifpluoeldx7zzglxcljo6oja4vrmtj7432rphldpdmm2